src/shared/utils/util.ts

```diff
@@ -425,8 +426,23 @@ export class Util {
     return new Promise((resolve) => setTimeout(resolve, ms));
   }
 
-  static removeNullFields(entity: any): void {
-    Object.keys(entity).forEach((k) => entity[k] == null && delete entity[k]);
+  static removeNullFields<T extends object>(entity: T): Partial<T> {
+    const result = { ...entity };
+    Object.keys(result).forEach((k) => result[k] == null && delete result[k]);
+    return result;
+  }
```

Comment:
Why not the same implementation as in services (using `Object.fromEntries`)? 😉  And maybe also allow undefined  entities => return entity?